app/views/cycle.js

```diff
@@ -0,0 +1,65 @@
+import document from "document";
+
+/**
+ * Cycle views menu entry
+ */
+
+const NUM_CYCLE_VIEWS = 3;
+
+let cycleCurrent;
+let cycleViewsLoaded;
+
+function cyclePrev() {
+  if (cycleCurrent === 0) {
+    return;
+  }
+
+  cycleCurrent--;
+
+  document.history.back();
+}
+
+function cycleNext() {
+  if (cycleCurrent === NUM_CYCLE_VIEWS - 1) {
+    return;
+  }
+
+  cycleCurrent++;
+
+  /**
+   * Lazy initialize the view. Note that event handlers are invalidated after when calling
+   * `document.location.assign()`, but will be set back in `updateCycleViews()`
+   */
+  if (cycleViewsLoaded[cycleCurrent] === false) {
+    document.location.assign(`cycle${cycleCurrent}.view`).then(updateCycleViews);
+    cycleViewsLoaded[cycleCurrent] = true;
+  } else {
+    document.history.forward();
+  }
+}
+
+function updateCycleViews() {
+  document.getElementById("btn-prev").addEventListener("click", () => {
+    console.log("btn-prev clicked");
+    cyclePrev();
+  });
+
+  document.getElementById("btn-next").addEventListener("click", () => {
+    console.log("btn-next clicked");
+    cycleNext();
+  });
+
+  document.onbeforeunload = (evt) => {
+    console.log("onbeforeunload called");
+    evt.preventDefault();
+    document.history.go(-cycleCurrent - 1);
+    document.onbeforeunload = undefined;
+  }
+}
+
+export function init() {
+  console.log("cycle-views start");
+  cycleCurrent = 0;
+  cycleViewsLoaded = [true, false, false];
+  document.location.assign(`cycle${cycleCurrent}.view`).then(updateCycleViews);
```

Comment:
Can this call be de-duped with the one in `cycleNext`?

Reply:
Yep, fixed, will be visible in the next push. Thanks!